js/chatbot/shared/CopyAll.svelte

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+	import { onDestroy } from "svelte";
+	import { Copy, Check } from "@gradio/icons";
+	import type { NormalisedMessage } from "../types";
+
+	let copied = false;
+	export let value: NormalisedMessage[] | null;
+
+	let timer: NodeJS.Timeout;
+
+	function copy_feedback(): void {
+		copied = true;
+		if (timer) clearTimeout(timer);
+		timer = setTimeout(() => {
+			copied = false;
+		}, 1000);
+	}
+
+	const copy_conversation = (): void => {
+		if (value) {
+			const conversation_text = value
+				.map((message) => `${message.role}: ${message.content}`)
```

Comment:
This is going to render images/videos as `[Object Object]`. I think the file url should be used instead. Or maybe some placeholder like `Generated File`